src/module_1/module_1_meteo_api.py

```diff
@@ -24,60 +25,68 @@ def call_API(url):
-    
+
     if 200 == response.status_code:
         return response.json()
-    else: print("Status code: ",response.status_code)
+    else:
+        print("Status code: ", response.status_code)
 
 
 def get_data_meteo_api(city):
-
     full_url = API_URL + "latitude=" + str(COORDINATES[city]["latitude"])
     full_url += "&longitude=" + str(COORDINATES[city]["longitude"])
     full_url += "&start_date=1950-01-01&end_date=2050-12-31"
-    full_url += "&models=CMCC_CM2_VHR4,FGOALS_f3_H,HiRAM_SIT_HR,MRI_AGCM3_2_S,EC_Earth3P_HR,MPI_ESM1_2_XR,NICAM16_8S"
+    full_url += "&models=CMCC_CM2_VHR4,FGOALS_f3_H,HiRAM_SIT_HR,MRI_AGCM3_2_S,EC_Earth3P_HR,MPI_ESM1_2_XR,NICAM16_8S" # noqa
     full_url += "&daily=" + VARIABLES
     return call_API(full_url)
 
 
 def transform_data(dataset):
+    dataset = dataset.dropna(axis=1, how="all")
+    dataset["Average"] = dataset.mean(axis=1, skipna=True, numeric_only=True)
+    dataset["Dispersion"] = dataset.std(axis=1, skipna=True, numeric_only=True)
 
-    dataset = dataset.dropna(axis=1, how='all')
-    dataset['Average'] = dataset.mean(axis=1, skipna=True, numeric_only=True)
-    dataset['Dispersion'] = dataset.var(axis=1, skipna=True, numeric_only=True)
-
-    dataset['time'] = pd.to_datetime(dataset['time'])
-    dataset['year'] = dataset['time'].dt.year
+    dataset["time"] = pd.to_datetime(dataset["time"])
+    dataset["year"] = dataset["time"].dt.year
 
     return dataset
-    
 
-def plot_data(data, yaxis):
 
-    yearly_average = data.groupby('year')['Average'].mean().reset_index()
-    yearly_variance = data.groupby('year')['Dispersion'].mean().reset_index()
+def plot_data(data, yaxis):
+    yearly_average = data.groupby("year")["Average"].mean().reset_index()
+    yearly_std = data.groupby("year")["Dispersion"].mean().reset_index()
 
     plt.figure(figsize=(10, 6))
-    sns.lineplot(data=yearly_average, x='year', y='Average', marker='o',
-                linestyle='-', label=yaxis)
-
-    plt.fill_between(yearly_variance['year'], 
-                    yearly_average['Average'] - yearly_variance['Dispersion'], 
-                    yearly_average['Average'] + yearly_variance['Dispersion'], 
-                    color='lightblue', alpha=0.5, label='Dispersion')
-
-    plt.xlabel('Year')
+    sns.lineplot(
+        data=yearly_average,
+        x="year",
+        y="Average",
+        marker="o",
+        linestyle="-",
+        label=yaxis,
+    )
+
+    plt.fill_between(
+        yearly_std["year"],
+        yearly_average["Average"] - yearly_std["Dispersion"],
+        yearly_average["Average"] + yearly_std["Dispersion"],
+        color="lightblue",
+        alpha=0.5,
+        label="Dispersion",
+    )
+
+    plt.xlabel("Year")
     plt.ylabel(yaxis)
     plt.title(yaxis)
     plt.legend()
     plt.grid(True)
 
     plt.show()
 
-def main():
 
+def main():
     json_file = get_data_meteo_api("Madrid")
-    df = pd.DataFrame(json_file['daily'])
+    df = pd.DataFrame(json_file["daily"])
```

Comment:
This will break if `status_code != 200` since `get_data_meteo_api` would return `None`. If you want any API failure to be the end of the street you should raise the exception within `call_API` rather than returning `None`.